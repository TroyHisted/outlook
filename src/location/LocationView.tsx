import React from 'react';
import Weather from '../weather/Weather';

export default (props: {
	getLocation: () => void;
	locationDescription: string;
	latitude: number;
	longitude: number;
}) => (
	<>
		<button
			onClick={props.getLocation}
			className="pure-button pure-button-primary"
		>
			Get Current Location
		</button>
		<p>Current location: {props.locationDescription}</p>
		{props.latitude !== 0 && props.longitude !== 0 && (
			<Weather latitude={props.latitude} longitude={props.longitude} />
		)}
	</>
);
