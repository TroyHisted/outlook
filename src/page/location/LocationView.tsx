import React from 'react';
import WeatherForecast from '../weather/WeatherForecast';

export default (props: {
	getLocation: () => void;
	locationDescription: string;
	latitude: number;
	longitude: number;
}) => (
	<>
		<button
			onClick={props.getLocation}
			className="pure-button pure-button-primary"
		>
			Get Current Location
		</button>
		<p>Current location: {props.locationDescription}</p>
		{props.latitude !== 0 && props.longitude !== 0 && (
			<WeatherForecast
				latitude={props.latitude}
				longitude={props.longitude}
			/>
		)}
	</>
);
