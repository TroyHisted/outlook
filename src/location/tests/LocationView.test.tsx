import renderer from 'react-test-renderer';
import LocationView from '../LocationView';

test('renders correctly', () => {
	const tree = renderer
		.create(
			<LocationView
				getLocation={() => {}}
				latitude={25}
				longitude={50}
				locationDescription="Nowhere"
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
