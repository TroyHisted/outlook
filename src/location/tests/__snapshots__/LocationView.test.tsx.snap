// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <button
    className="pure-button pure-button-primary"
    onClick={[Function]}
  >
    Get Current Location
  </button>,
  <p>
    Current location: 
    Nowhere
  </p>,
  <p>
    Current Weather: 
    unknown
  </p>,
]
`;
